src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php

```diff
@@ -213,4 +213,21 @@ private function hasAttributeAccessorMethod(string $class, string $attribute): b
             && !$method->getAttributes(Ignore::class)
             && !$method->getNumberOfRequiredParameters();
     }
+
+    private function isReadableCache(object|string  $class, object|string $classOrObject, string $attribute): bool
```

Comment:
can `$class` actually be an object ? AFAICT, the caller always pass a string there

Reply:
Yes, `$class` can be a string. So `object|string` is not correct.

I created a branch as I had it first, before I had committed the PR. Here the [Example](https://github.com/symfony/symfony/commit/719e49655d277639ccf6f6ec54da64d8b2294f9c).

But I think it iss more important that when there are too much adjustments, the condition is no longer on one line, but with line breaks. This is clearer and helps to avoid possible merge conflicts.

```php
if (!isset(self::$isReadableCache[$class.$attribute])) {
    self::$isReadableCache[$class.$attribute] = $this->propertyInfoExtractor->isReadable($class, $attribute)
        || $this->hasAttributeAccessorMethod($class, $attribute)
        || (\is_object($classOrObject) && $this->propertyAccessor->isReadable($classOrObject, $attribute));
}
```

Reply:
@stof Can you please give me some feedback?